src/control/server/harness.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"os"
+	"time"
+
+	srvpb "github.com/daos-stack/daos/src/control/common/proto/srv"
+	"github.com/daos-stack/daos/src/control/ioserver"
+	"github.com/pkg/errors"
+)
+
+type managedInstance struct {
+	cfg                      *ioserver.Config
+	instance                 *ioserver.Instance
+	superblock               *Superblock
+	shouldReformatSuperblock bool
+	fsRoot                   string
+}
+
+func (mi *managedInstance) Setup() (err error) {
+	mi.superblock, err = mi.ReadSuperblock()
+	if err == nil && !mi.shouldReformatSuperblock {
+		return nil
+	}
+
+	if !os.IsNotExist(errors.Cause(err)) {
```

Comment:
would this not resolve with err, does it need Cause(err)?

Reply:
Might, might not. If there was an additional layer of wrapping added somewhere then this test would not work properly. Using errors.Cause() guarantees that we're inspecting the original error.